tests/multi_bluetooth/stress_log_filesystem.py

```diff
@@ -0,0 +1,185 @@
+_IRQ_GATTC_CHARACTERISTIC_RESULT = const(11)
+_IRQ_GATTC_CHARACTERISTIC_DONE = const(12)
+_IRQ_GATTC_READ_RESULT = const(15)
+_IRQ_GATTC_READ_DONE = const(16)
+_IRQ_GATTC_WRITE_DONE = const(17)
+
+SERVICE_UUID = bluetooth.UUID("A5A5A5A5-FFFF-9999-1111-5A5A5A5A5A5A")
+CHAR_UUID = bluetooth.UUID("00000000-1111-2222-3333-444444444444")
+CHAR = (
+    CHAR_UUID,
+    bluetooth.FLAG_READ | bluetooth.FLAG_WRITE | bluetooth.FLAG_NOTIFY | bluetooth.FLAG_INDICATE,
+)
+SERVICE = (
+    SERVICE_UUID,
+    (CHAR,),
+)
+SERVICES = (SERVICE,)
+
+
+waiting_events = {}
+log_file = None
+
+def write_log(*args):
+    if log_file:
+        print(*args, file=log_file)
+        log_file.flush()
+
+last_file_write = 0
+
+def periodic_log_write():
+    global last_file_write
+    t = time.ticks_ms()
+    if time.ticks_diff(t, last_file_write) > 50:
+        write_log('tick')
+        last_file_write = t
+
+
+def irq(event, data):
+    write_log('event', event)
+
+    if event == _IRQ_CENTRAL_CONNECT:
+        print("_IRQ_CENTRAL_CONNECT")
+        waiting_events[event] = data[0]
+    elif event == _IRQ_CENTRAL_DISCONNECT:
+        print("_IRQ_CENTRAL_DISCONNECT")
+    elif event == _IRQ_PERIPHERAL_CONNECT:
+        print("_IRQ_PERIPHERAL_CONNECT")
+        waiting_events[event] = data[0]
+    elif event == _IRQ_PERIPHERAL_DISCONNECT:
+        print("_IRQ_PERIPHERAL_DISCONNECT")
+    elif event == _IRQ_GATTC_SERVICE_RESULT:
+        # conn_handle, start_handle, end_handle, uuid = data
+        if data[-1] == SERVICE_UUID:
+            print("_IRQ_GATTC_SERVICE_RESULT", data[3])
+            waiting_events[event] = (data[1], data[2],)
+        else:
+            return
+    elif event == _IRQ_GATTC_SERVICE_DONE:
+        print("_IRQ_GATTC_SERVICE_DONE")
+    elif event == _IRQ_GATTC_CHARACTERISTIC_RESULT:
+        # conn_handle, def_handle, value_handle, properties, uuid = data
+        if data[-1] == CHAR_UUID:
+            print("_IRQ_GATTC_CHARACTERISTIC_RESULT", data[-1])
+            waiting_events[event] = data[2]
+        else:
+            return
+    elif event == _IRQ_GATTC_CHARACTERISTIC_DONE:
+        print("_IRQ_GATTC_CHARACTERISTIC_DONE")
+    elif event == _IRQ_GATTC_READ_RESULT:
+        print("_IRQ_GATTC_READ_RESULT", bytes(data[-1]))
+    elif event == _IRQ_GATTC_READ_DONE:
+        print("_IRQ_GATTC_READ_DONE", data[-1])
+    elif event == _IRQ_GATTC_WRITE_DONE:
+        print("_IRQ_GATTC_WRITE_DONE", data[-1])
+    elif event == _IRQ_GATTS_WRITE:
+        print("_IRQ_GATTS_WRITE")
+    elif event == _IRQ_GATTS_READ_REQUEST:
+        print("_IRQ_GATTS_READ_REQUEST")
+
+    if event not in waiting_events:
+        waiting_events[event] = None
+
+
+def wait_for_event(event, timeout_ms):
+    t0 = time.ticks_ms()
+    while time.ticks_diff(time.ticks_ms(), t0) < timeout_ms:
+        periodic_log_write()
+        if event in waiting_events:
+            result = waiting_events[event]
+            del waiting_events[event]
```

Comment:
I think `waiting_events.pop(event)` is a better way to write this, it's simpler and more atomic (especially if IRQs can change this dict).

Could even write this as:
```python
if result := waiting_events.pop(event, -1) != -1:
    return result
```
But not sure if it's good to put a dependency on the walrus operator....